### PR TITLE
Doc: Dedicated server dependencies

### PR DESCRIPTION
**Pull request by Berbe**
## Motivation / Problem
Better document dedicated servers usage

## Description
Following #8802, enhance the documentation of the requirements

### CODE REVIEW

**Comment by LordAro** [comment]
I think expanding/improving the comment at the bottom of the list would be more beneficial

**Review by LordAro — DISMISSED** [review]

**Comment by TrueBrain** [comment]
If we are fiddling with this, we might as well make it more correct in general. Give me a few to cook up what I mean :)

File: COMPILING.md [hunk 1]
```diff
@@ -110,11 +111,14 @@ the base sets which are not yet included in the base set information files.
...
 (`GRFCODEC_EXECUTABLE` and `NFORENUM_EXECUTABLE`).
 
-## Developers settings
-
+## Compilation settings
 You can control some flags directly via `CXXFLAGS` (any combination
 of these flags will work fine too):
 
+### Dedicated server
+- `-DOPTION_DEDICATED=ON`: compiles the dedicated server version, without libraries unneeded in a headless environment
```
Comment (TrueBrain) [inline COMPILING.md:119]: Not only that I am rewriting some other parts I noticed this is wrong. This chapter is about `CXXFLAGS`, while `-DOPTION_DEDICATED` is a CMake flag :) I have a PR incoming that also includes this PR, fixing this :)

**Comment by TrueBrain** [comment]
Ended up creating a new PR, sorry about that. But it seems we were slacking heavily on keeping this correct and up-to-date. Let me know if you like my solution, and feel free to add better suggestions :D

**Comment by Berbe** [comment]
Alrighty, then: closing in favour of #9074

**Pull request closed without merging**